src/engine/ExportQueryExecutionTrees.cpp

```diff
@@ -518,6 +518,37 @@ ExportQueryExecutionTrees::idToLiteral(const Index& index, Id id,
   }
 }
 
+// _____________________________________________________________________________
+std::optional<ad_utility::triple_component::LiteralOrIri>
+ExportQueryExecutionTrees::idToLiteralOrIri(const Index& index, Id id,
+                                            const LocalVocab& localVocab) {
+  using enum Datatype;
+  switch (id.getDatatype()) {
+    case WordVocabIndex:
+      return LiteralOrIri{
+          ad_utility::triple_component::Literal::literalWithoutQuotes(
+              index.indexToString(id.getWordVocabIndex()))};
+    case VocabIndex:
+    case LocalVocabIndex:
+      return getLiteralOrIriFromVocabIndex(index, id, localVocab);
+    case TextRecordIndex:
+      return LiteralOrIri{
+          ad_utility::triple_component::Literal::literalWithoutQuotes(
+              index.getTextExcerpt(id.getTextRecordIndex()))};
+    default:
+      auto idLiteralAndType = idToStringAndTypeForEncodedValue(id);
+      if (idLiteralAndType.has_value()) {
+        auto lit = ad_utility::triple_component::Literal::literalWithoutQuotes(
+            idLiteralAndType.value().first);
+        lit.addDatatype(
+            ad_utility::triple_component::Iri::fromIrirefWithoutBrackets(
+                idLiteralAndType.value().second));
+        return LiteralOrIri{lit};
+      }
```

Comment:
Also factor this out (and possibly reuse it in other places) , 
`idToLiteralOrIriForEncodedValue`.